module_utils/network_lsr/argument_validator.py

```diff
@@ -1964,6 +1975,12 @@ def _ipv4_enabled(connection):
                     "Setting 'dns_search' or 'dns_options' is not allowed when "
                     "both IPv4 and IPv6 are disabled.",
                 )
+            elif not _ipv4_enabled(connection) and _ipv6_ignore(connection):
```

Comment:
for better or worse, NetworkManager does [this](https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/blob/6b3862e39ac4e2dee19d6b87c117ea3a1125c9dd/src/libnm-core-impl/nm-setting-ip6-config.c#L226). Meaning, it rejects setting "dns-search" for methods "ignore", "link-local" and "disabled".

The check `_ipv6_ignore()` only checks for the first of those 3 cases (as the name of the function indicates -- whereas the role currently never sets `ipv6.method=link-local`).